pcap/index.go

```diff
@@ -21,40 +21,72 @@ type Section struct {
 	Index  ranger.Envelope
 }
 
-// CreateIndex creates an index for a pcap file.  If the file isn't
-// a pcap file, an error is returned.
-func CreateIndex(r io.Reader, limit int) (Index, error) {
-	reader, err := pcapio.NewPcapReader(r) // XXX TBD: lookup the right reader
-	if err != nil {
-		return nil, err
-	}
+// CreateIndex creates an index for a pcap file.
+func CreateIndex(reader pcapio.Reader, limit int) (Index, error) {
```

Comment:
Unless necessary I would be obliged to keep this interface the same- more boiler plate for clients.

Reply:
excellent point.  will fix.